pandas/tests/test_base.py

```diff
@@ -1273,3 +1274,48 @@ def test_to_numpy_dtype(as_series):
     expected = np.array(['2000-01-01T05', '2001-01-01T05'],
                         dtype='M8[ns]')
     tm.assert_numpy_array_equal(result, expected)
+
+
+def check_pinned_names(cls):
+    """
+    Check that the any dynamically-defined methods have the correct
+    names, i.e. not 'wrapper'.
+    """
+    special_cases = {
+        "isnull": "isna",
+        "notnull": "notna",
+        "take": "take_nd",     # Categorical
+        "to_list": "tolist",   # Categorical
+        "iteritems": "items",
+        "__bool__": "__nonzero__",
+    }
+    ignore = {
+        "_create_comparison_method",
+    }
+    if 'Subclassed' in cls.__name__:
+        # dummy classes defined in tests
+        return
+    for name in dir(cls):
```

Comment:
This iteration looks heavy. What time does it take to run through all cases defined in the test_pinned_names parametrization? Should test_pinned_names below get a pytest slow marker?

Would it not be better to do specific tests for the cases where methods are generated dynamically instead of this broader test?

Reply:
> What time does it take to run through all cases defined in the test_pinned_names parametrization? Should test_pinned_names below get a pytest slow marker?

Locally `python -m pytest pandas/tests/test_base.py -k test_pinned_names` takes 1.02 seconds.

> Would it not be better to do specific tests for the cases where methods are generated dynamically instead of this broader test?

No, that's an invitation to have things fall through the cracks down the road.